lib/daru/dataframe.rb

```diff
@@ -1419,6 +1422,29 @@ def report_building(b) # :nodoc: #
         end
       end
     end
+    #custom function for dataframe summary
+    def g_summary()
+       frame_summary = "= \t\n"
+       frame_summary << "\tNumber of rows: #{nrows}\n"
+       @raw = self
+       #no_of_vect = self.vectors.count
+       
+       @raw.each do |v|
+        frame_summary << "\tElement:[#{v.name}]\n"
+        frame_summary << "\t== #{v.name}\n"
+        count = v.size
+        valid_count = 0
+        v.to_a.each do |val_n|
+           if (val_n!=nil && !(val_n.is_a?(Float) && val_n.nan?))
+              valid_count = valid_count + 1 
+           end
+        end
+       
+        frame_summary << "\t  n : #{count}\n"
+        frame_summary << "\t  n valid : #{valid_count}\n"
+       end
```

Comment:
You should go through the code of the report builder gem and see ALL  the data that it outputs. It exists for more reasons than just telling `n` and `n_valid`.

The patch you submit must emulate that output as much as possible. Try running reportbuilder through different examples and see their respective outputs.